cpmpy/expressions/core.py

```diff
@@ -153,8 +153,8 @@ def __repr__(self):
         return "{}({})".format(self.name, ",".join(strargs))
 
     def __hash__(self):
-        return hash(self.__repr__())
-
+        args = flatlist(self.args)
```

Comment:
I'm a bit hesitant for this... you are first creating a list of list of lists, then turning that into a tuple (single tuple? or tuple of tuples?)

I think we should have something that creates tuples directly? also if its ndarray we can call its custom flatten?